src/components/navigation/navigation.js

```diff
@@ -1,16 +1,33 @@
 import './navigation.css'
-import {Link} from 'react-router-dom'
+import {Link, useNavigate} from 'react-router-dom'
 import { useProductContext } from '../../contexts/product-context';
 import {useWishContext} from '../../contexts/wishlist-context';
-import {useCartContext} from "../../contexts/cart-context"
+import {useCartContext} from "../../contexts/cart-context";
+import {useAuthContext} from "../../contexts/auth-context"
 
 function Navigation() {
+  const navigate = useNavigate()
+  const {authCred, setAuthCred} = useAuthContext()
   const {dispatch} = useProductContext()
   const {wishState} = useWishContext()
   const {cartState} = useCartContext()
   const {cartProducts} = cartState
-  const qunatityReducer = (previousProd, currentProd) => currentProd.quantity + previousProd
+  const qunatityReducer = (previousProd, currentProd) => currentProd.qty + previousProd
   const cartProductQuantity = cartProducts.reduce(qunatityReducer, 0)
+
+  const navNavigation = (type) => {
+    type === "wishlist" ? (authCred.authStatus ? (navigate('/wishlist')):(navigate('/login'))) : (authCred.authStatus ? (navigate('/cart')):(navigate('/login'))) 
```

Comment:
follow a consistent naming convention for type, here it is small letters & at few places you have used CAPS